crates/holochain_cascade/src/lib.rs

```diff
@@ -615,6 +636,17 @@ where
         Ok(results)
     }
 
+    #[instrument(skip(self, options))]
+    /// Takes ownership of `self` to perform a `dht_get_links`.
+    /// Useful for async closures tht will drop a `&mut self` reference before the async completes.
+    pub async fn into_dht_get_links(
+        mut self,
+        key: WireLinkKey,
+        options: GetLinksOptions,
+    ) -> CascadeResult<Vec<Link>> {
+        self.dht_get_links(key, options).await
+    }
```

Comment:
I think this should be solved on the closure level by cloning the cascade not adding extra api to the cascade

Reply:
@freesig you would think so, but note the `Cascade` e.g. in `get.rs` is already owned so cloning won't do anything:

```rust
                        Cascade::from_workspace_network(
                            call_context.host_context.workspace(),
                            call_context.host_context.network().clone()
                        )
                        .into_dht_get(any_dht_hash, get_options)
```

so the error for the clone case looks like this:


```rust
error[E0515]: cannot return value referencing temporary value
  --> crates/holochain/src/core/ribosome/host_fn/get.rs:24:25
   |
24 |                            Cascade::from_workspace_network(
   |   _________________________^
   |  |_________________________|
   | ||
25 | ||                             call_context.host_context.workspace(),
26 | ||                             call_context.host_context.network().clone()
27 | ||                         )
28 | ||                         .clone()
   | ||________________________________- temporary value created here
29 | |                          .dht_get(any_dht_hash, get_options)
   | |____________________________________________________________^ returns a value referencing data owned by the current function
```